src/token-group/dismiss-button.tsx

```diff
@@ -6,20 +6,24 @@ import InternalIcon from '../icon/internal';
 
 interface DismissButtonProps {
   disabled?: boolean;
+  readOnly?: boolean;
   onDismiss?: () => void;
   dismissLabel?: string;
 }
 
 export default forwardRef(DismissButton);
 
-function DismissButton({ disabled, dismissLabel, onDismiss }: DismissButtonProps, ref: Ref<HTMLButtonElement>) {
+function DismissButton(
+  { disabled, dismissLabel, onDismiss, readOnly }: DismissButtonProps,
+  ref: Ref<HTMLButtonElement>
+) {
   return (
     <button
       ref={ref}
       type="button"
       className={styles['dismiss-button']}
       aria-disabled={disabled ? true : undefined}
-      onClick={!disabled && onDismiss ? () => onDismiss() : undefined}
+      onClick={!disabled && !readOnly && onDismiss ? () => onDismiss() : undefined}
```

Comment:
Non-blocking: Just noticing how inconsistent we are in code for handling this. Some cases we return and then like here and a few other places we assign undefined.

Reply:
Changed to

`onClick={() => {
  if (disabled || readOnly || !onDismiss) {
    return;
  }

  onDismiss();
}}`